### DISCUSSION TITLE
Support for user groups and assigning roles to them apart from directly to users

### DISCUSSION THREAD

**Discussion by tassosgeo** [apache/superset#18556] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently superset only support user roles and not also user groups.
This will be inline with most openid authentication providers so that groups and roles can propagate from them to superset.
Currently both of them have to be mapped as roles in superset.

**Describe the solution you'd like**
To be able to create user groups and assign roles to these groups, and not only assign roles to users directly

**Comment by giovannipapini-agilelab** [comment]
Is there any update about this feature?

**Comment by frabenetti** [comment]
This functionality would be a great improvement, is there any news about it?

**Comment by howardcho** [comment]
I'm also very interested in this, as it would make user management much easier.

**Comment by shauryagoel** [comment]
Is there any update on this? 
@sfirke

**Reply by sfirke** [reply]
It seems like a good feature request to me to.  I don't know of anyone working on this and it's not on the (still-evolving) [list of ideas for 4.0](https://github.com/orgs/apache/projects/292).  If someone wants to try to contribute this, please propose a SIP (Superset Improvement Proposal) to hammer out the details of your proposed implementation.
